app/src/main/java/com/example/bigowlapp/utils/AuthFailureNotificationListener.java

```diff
@@ -0,0 +1,66 @@
+package com.example.bigowlapp.utils;
+
+
+import android.content.Context;
+import android.widget.Toast;
+
+import androidx.annotation.Nullable;
+
+import com.example.bigowlapp.repository.RepositoryFacade;
+import com.google.firebase.Timestamp;
+import com.google.firebase.firestore.DocumentChange;
+import com.google.firebase.firestore.EventListener;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.firestore.FirebaseFirestoreException;
+import com.google.firebase.firestore.QuerySnapshot;
+
+import static com.google.firebase.firestore.DocumentChange.Type.ADDED;
+
+public class AuthFailureNotificationListener {
+
+    private final FirebaseFirestore db = FirebaseFirestore.getInstance();
+    private static final int THIRTY_MINUTES = 1800;
+    private SmsSender smsSender = new SmsSender();
+    RepositoryFacade repositoryFacade = RepositoryFacade.getInstance();
+
+
+    public AuthFailureNotificationListener() {
+
+    }
+
+    public void listen(Context context) {
+        db.collection("notifications")
+                //.whereEqualTo("receiverUid", repositoryFacade.getAuthRepository().getCurrentUser().getUid())
+                .addSnapshotListener(new EventListener<QuerySnapshot>() {
```

Comment:
new EventListener<QuerySnapshot>() is the old way, try replacing it with lambda. With that, you don't even have to include @Override tag and the overriding method! In your case, the lambda would be "(snapshots, e) -> "

Reply:
Yes, I also saw the IDE suggestion about that. I did that